rust/ql/test/query-tests/security/CWE-328/test.rs

```diff
@@ -11,8 +11,8 @@ fn test_hash_algorithms(
 
     // MD5
     _ = md5::Md5::digest(harmless);
-    _ = md5::Md5::digest(credit_card_no); // $ Alert[rust/weak-sensitive-data-hashing]
-    _ = md5::Md5::digest(password); // $ Alert[rust/weak-sensitive-data-hashing]
+    _ = md5::Md5::digest(credit_card_no); // $ MISSING: Alert[rust/weak-sensitive-data-hashing]
```

Comment:
Currently `getStaticTarget()` fails on `md5::Md5::digest`. This are the relevant lines:

```rust
pub type Md5 = CoreWrapper<Md5Core>;

impl<D: FixedOutput + Default + Update + HashMarker> Digest for D {
    fn digest(data: impl AsRef<[u8]>) -> Output<Self> { ... }
}
```

* https://github.com/RustCrypto/hashes/blob/md-5-v0.10.6/md5/src/lib.rs#L134
* https://github.com/RustCrypto/traits/blob/digest-v0.10.7/digest/src/digest.rs#L57

Reply:
@hvitved I guess the type variable in `for D` is the cause of the trouble, isn't it?

Reply:
Yes, we don't currently handle `impl<T> Foo for T`; I have added this instance to our internal issue tracking this.